### PR TITLE
Add TimeTracking to Gateway 

### PR DESCRIPTION
**Pull request by JBHarvey**

### CODE REVIEW

File: sensor-gateway/common/data-structure/spirit/Frame.h [hunk 1]
```diff
@@ -29,7 +32,7 @@ namespace DataFlow {
 
     public:
 
-        explicit Frame(FrameId frameId, SystemId systemId, PixelsArray pixels);
+        explicit Frame(MessageId messageId, SystemId systemId, PixelsArray pixels);
```
Comment (Sebdevar) [inline sensor-gateway/common/data-structure/spirit/Frame.h]: Should probably refactor the class' name from Message to Frame at this point.

File: sensor-gateway/common/data-structure/spirit/Pixel.h [hunk 2]
```diff
@@ -81,7 +81,13 @@ namespace Defaults {
         using DataFlow::Pixel;
         using DataFlow::PixelId;
         using DataFlow::TracksArray;
-        PixelId const UNDEFINED_ID = std::numeric_limits<PixelId>::infinity();
+
+        /**
+         * @warning Modifying these values without updating the FlatBuffers file WILL create a huge performance drop!!!
+         * If a value is modified here, be sure it homologous value in the FlatBuffers schema file.
```
Comment (Sebdevar) [inline sensor-gateway/common/data-structure/spirit/Pixel.h]:
```suggestion
         * If a value is modified here, be sure its corresponding value in the FlatBuffers schema file stays homologous.
```

File: sensor-gateway/data-translation/AWLTranslationStrategy.cpp [hunk 3]
```diff
@@ -54,10 +54,10 @@ void AWLTranslationStrategy::translateRawData(SensorRawData&& serverRawData) {
 }
 
 void AWLTranslationStrategy::translateEndOfFrameMessage(SensorMessage&& sensorMessage) {
-    FrameId frameId = convertTwoBytesToUnsignedBigEndian(sensorMessage.data[0], sensorMessage.data[1]);
+    MessageId frameId = convertTwoBytesToUnsignedBigEndian(sensorMessage.data[0], sensorMessage.data[1]);
```
Comment (Sebdevar) [inline sensor-gateway/data-translation/AWLTranslationStrategy.cpp]:
```suggestion
    MessageId messageId = convertTwoBytesToUnsignedBigEndian(sensorMessage.data[0], sensorMessage.data[1]);
```

File: sensor-gateway/data-translation/AWLTranslationStrategy.cpp [hunk 4]
```diff
@@ -54,10 +54,10 @@ void AWLTranslationStrategy::translateRawData(SensorRawData&& serverRawData) {
...
     SystemId systemId = convertTwoBytesToUnsignedBigEndian(sensorMessage.data[2], sensorMessage.data[3]);
     currentOutputMessage.systemId = systemId;
-    currentOutputMessage.frameId = frameId;
+    currentOutputMessage.messageId = frameId;
```
Comment (Sebdevar) [inline sensor-gateway/data-translation/AWLTranslationStrategy.cpp]:
```suggestion
    currentOutputMessage.messageId = messageId;
```

File: sensor-gateway/data-translation/GuardianTranslationStrategy.cpp [hunk 5]
```diff
@@ -60,10 +60,10 @@ void GuardianTranslationStrategy::translateRawData(SensorRawData&& sensorRawData
 }
 
 void GuardianTranslationStrategy::translateEndOfFrameMessage(SensorMessage&& sensorMessage) {
-    FrameId frameId = convertTwoBytesToUnsignedBigEndian(sensorMessage.data[0], sensorMessage.data[1]);
+    MessageId frameId = convertTwoBytesToUnsignedBigEndian(sensorMessage.data[0], sensorMessage.data[1]);
```
Comment (Sebdevar) [inline sensor-gateway/data-translation/GuardianTranslationStrategy.cpp]:
```suggestion
    MessageId messageId = convertTwoBytesToUnsignedBigEndian(sensorMessage.data[0], sensorMessage.data[1]);
```

File: sensor-gateway/data-translation/GuardianTranslationStrategy.cpp [hunk 6]
```diff
@@ -60,10 +60,10 @@ void GuardianTranslationStrategy::translateRawData(SensorRawData&& sensorRawData
...
     SystemId systemId = convertTwoBytesToUnsignedBigEndian(sensorMessage.data[2], sensorMessage.data[3]);
     currentOutputMessage.systemId = systemId;
-    currentOutputMessage.frameId = frameId;
+    currentOutputMessage.messageId = frameId;
```
Comment (Sebdevar) [inline sensor-gateway/data-translation/GuardianTranslationStrategy.cpp]:
```suggestion
    currentOutputMessage.messageId = messageId;
```

File: sensor-gateway/common/TypeDefinition.h [hunk 7]
```diff
@@ -75,7 +75,7 @@ namespace DataFlow {
     typedef uint16_t Intensity;
     typedef uint16_t PixelId;
     typedef int16_t Speed;
-    typedef uint16_t SystemId;
+    typedef uint16_t SensorId;
```
Comment (Sebdevar) [inline sensor-gateway/common/TypeDefinition.h]: Why the switch from System (the agreed-upon name (*or so I though*)) to Sensor?

File: test/utilities/files/FrameFileManager.h [hunk 8]
```diff
@@ -55,7 +58,7 @@ namespace TestUtilities {
 
         std::string const TRACK_ID_LABEL = "Track id";
 
-        std::string const INTENSITY_LABEL = "Ìntensity";
```
Comment (Sebdevar) [inline test/utilities/files/FrameFileManager.h]: lol, how that accent get there >.<

File: test/test-suite/common/data-structure/TimeTrackingTest.cpp [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+//    TimeTracking frame;
+//    Track track;
+//    auto expectedAddedTrack = track;
+//    auto trackPosition = 0;
+//
+//    frame.addTrackToPixelWithId(SOME_PIXEL_ID, std::move(track));
+//
+//    auto actualAddedTrack = frame.getPixels()->at(SOME_PIXEL_ID).getTracks()->at(trackPosition);
+//    ASSERT_EQ(expectedAddedTrack, actualAddedTrack);
+}
```
Comment (Sebdevar) [inline test/test-suite/common/data-structure/TimeTrackingTest.cpp]: Dead test?

Comment (JBHarvey) [reply test/test-suite/common/data-structure/TimeTrackingTest.cpp]: Non @Sebdevar  arrete de lire les trucs commit par commit pi tu tomberas plus sur du stock qui a ete remplace plus tard

**Pull request merged**